File: python/task11.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - #
# Module of tasks Hackerrank
# Copyright (C) 2021
# Salavat Bibarsov <Bibarsov.Salavat@gmail.com>
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #

# https://www.hackerrank.com/challenges/python-string-split-and-join/problem

"""
Task
You are given a string. Split the string on a " " (space) delimiter and join 
using a - hyphen.
Complete the split_and_join function in the editor below.

split_and_join has the following parameters:

string line: a string of space-separated words
"""


def split_and_join(line: str):
    
    splitted = line.split(' ')

    return '-'.join(splitted)

if __name__ == '__main__':

    line = input()
    result = split_and_join(line)

    print(result)

File: c/task11.c
#include <stdio.h>

// Bitwise Operators
// https://www.hackerrank.com/challenges/bitwise-operators-in-c/problem

/*
 * Complete the calculate_the_maximum function in the editor below.
 * calculate_the_maximum has the following parameters:
 * int n: the highest number to consider
 * int k: the result of a comparison must be lower than this number to be considered
*/

void calculate_the_maximum(int n, int k) {

    int and = 0, or = 0, xor = 0;

    for (int i = 1; i <= n; i++){

        for (int j = i+1; j <= n; j++){

            (k > (i & j)) && ((i & j) > and) ? and = (i & j) : 0;
            (k > (i | j)) && ((i | j) > or) ? or = (i | j) : 0;
            (k > (i ^ j)) && ((i ^ j)> xor) ? xor = (i ^ j) : 0;

        }

    }

    printf("%d\n%d\n%d\n", and, or, xor);
}

int main() {
    int n, k;

    scanf("%d %d", &n, &k);
    calculate_the_maximum(n, k);

    return 0;
}
